pkg/controller/servicebindingrequest/mapper.go

```diff
@@ -21,17 +120,60 @@ func (m *sbrRequestMapper) Map(obj handler.MapObject) []reconcile.Request {
 		"Object.Namespace", obj.Meta.GetNamespace(),
 		"Object.Name", obj.Meta.GetName(),
 	)
-	toReconcile := []reconcile.Request{}
 
-	sbrNamespacedName, err := getSBRNamespacedNameFromObject(obj.Object)
+	namespacedNamesToReconcile := make(namespacedNameSet)
+
+	if isServiceBindingRequest(obj.Object) {
+		requests := []reconcile.Request{
+			{NamespacedName: convertToNamespacedName(obj.Meta)},
+		}
+		log.Debug("current resource is a SBR", "Requests", requests)
+		return requests
+	}
+
+	sbrList, err := m.client.Resource(groupVersion).List(metav1.ListOptions{})
 	if err != nil {
-		log.Error(err, "on inspecting object for annotations for SBR object")
-		return toReconcile
+		log.Error(err, "listing SBRs")
+		// NOTE: nothing to do anymore if obj isn't a SBR and there's an error listing SBRs; I'm not
+		// sure whether retrying in the same reconcile loop is the proper approach so we'll rely on
+		// the resync period to re-process all watched objects.
+		return []reconcile.Request{}
```

Comment:
Perhaps we can use this function https://github.com/redhat-developer/service-binding-operator/blob/master/pkg/controller/servicebindingrequest/common.go#L45

Reply:
I believe you're referring to the following function:

https://github.com/redhat-developer/service-binding-operator/blob/ffbf35ff6f419d09795389a820fee61954abfdc6/pkg/controller/servicebindingrequest/common.go#L45-L50

The mapper might return reconciliation requests which will be queued to be processed by a worker; in othe words, the `Reconcile()` method is only executed after the mapping happens and produces some result.

Now that I'm thinking, that comment is misleading since it is fine if a resource is examined and the mapper results in an empty slice. 

Take the following use cases:

* Add a Service Binding Request in a namespace without declaring the `applicationSelector` field.
* Add a Deployment (the mapper will not find a relationship between the given resource and a SBR, so no SBRs will be further enqueued).
* Modify the Service Binding Request and modify the `applicationSelector` to include the newly added Deployment (this will trigger the SBR's reconciliation process).

The other use case would be create a Service Binding Request declaring `applicationSelector` with a missing application:

* Add a Service Binding Request in a namespace declaring the `applicationSelector` field, but no resource as of yet.
* Add a Deployment (the relationship will be identified by the mapper and the related SBR is enqueued).

Given this I think the behavior is fine, what do you think if I just amend the comment?

Reply:
Inspecting further the [`List` method][1], there are a couple of error types being returned, although from a developer's perspective I believe this section should be resilient mostly to `*errors.StatusError`, at least initially. 

Some sub-errors could be retried in the mapping procedure, such as `TooManyRequests` and `Timeout`, where others that are not recoverable such as `Forbidden` or `NotFound`, so it is likely this block should take this in account.



[1]: https://github.com/isutton/service-binding-operator/blob/ba129432e43bf26455edeebf88bcaf92184586c3/vendor/k8s.io/client-go/dynamic/simple.go#L260-L282

Reply:
I've learned the client is [programmed to retry][1] in specific situations the API server instructs it to; this makes me believe all errors at this point are not recoverable. I'll modify the comment to make that clear.

[1]: https://github.com/isutton/service-binding-operator/blob/e17445570bd3889bcf7499142350a3b81463c6be/vendor/k8s.io/client-go/rest/request.go#L723-L812